src/mpid/ch4/netmod/ucx/ucx_types.h

```diff
@@ -53,11 +51,12 @@ extern ucp_generic_dt_ops_t MPIDI_UCX_datatype_ops;
 
 #define MPIDI_UCX_CONTEXT_TAG_BITS 16
 #define MPIDI_UCX_CONTEXT_RANK_BITS 16
-#define UCX_TAG_BITS 32
+#define UCX_TAG_BITS 30
 
-#define MPIDI_UCX_TAG_MASK      (0x00000000FFFFFFFFULL)
-#define MPIDI_UCX_SOURCE_MASK   (0x0000FFFF00000000ULL)
-#define MPIDI_UCX_TAG_SHIFT     (32)
+#define MPIDI_UCX_TAG_MASK      (0x000000007FFFFFFFULL)
+#define MPIDI_UCX_SOURCE_MASK   (0x00007FFF80000000ULL)
```

Comment:
@raffenet I think this might work. I have moved the AM bit from the tag to the MSB in the matching bits, thus I had to change the bit masks.

Reply:
Can you update lines 48-50 with the new layout?

Reply:
Done

Reply:
OK, one last dumb question. The layout indicates 31 tag bits, but `UCX_TAG_BITS` is 30?

Reply:
The number of effective tag bits is 30. The 31st is reserved for some reason I suppose. In fact, the error bit is always the second MSB of the tag field not the first. Probably the way I am assigning the number of tag bits is confusing. Earlier on the `UCX_TAG_BITS` constant was assigned but unused and the number of actual tag bits was 31 + the AM bit.

Reply:
Actually forget about it, you are right is 31. Let me fix it